src/api.cc

```diff
@@ -573,12 +573,61 @@ CheckInResult AkliteClient::CheckInLocal(const LocalUpdateSource* local_update_s
     return CheckInResult(CheckInResult::Status::NoTargetContent, hw_id_, std::vector<TufTarget>{});
   }
 
-  if (!usingUpdateClientApi) {
+  if (invoke_post_cb_at_checkin_) {
     client_->notifyTufUpdateFinished();
   }
   return CheckInResult(check_status, hw_id_, toTufTargets(available_targets));
 }
 
+CheckInResult AkliteClient::CheckStoredMetadata(const LocalUpdateSource* local_update_source) const {
+  std::string err_msg;
+  LOG_INFO << "Checking the stored TUF metadata...";
+  try {
+    tuf_repo_->CheckMeta();
+    LOG_INFO << "The stored TUF metadata is valid";
+  } catch (const std::exception& exc) {
+    err_msg = std::string("Stored TUF metadata is invalid: ") + exc.what();
+    LOG_WARNING << err_msg;
+    return CheckInResult{CheckInResult::Status::SecurityError, hw_id_, {}};
+  }
+
+  LOG_INFO << "Searching for matching TUF Targets...";
```

Comment:
Optional, not necessary at this stage/PR. The code after CheckMeta() seems similar or even identical to the one we have in CheckIn. So, we may wanna consider unifying it, e.g. moving to the common/shared private method, etc.

Reply:
I agree. I was going to do it right at the beginning, but the amount of changes required in the CheckIn / CheckInLocal methods made me postpone it. Perhaps I'll do it as a separate patch.

Reply:
No worries, it is not critical at all.